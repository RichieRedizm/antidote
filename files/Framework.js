;(global => {
	var Framework = input => {
		return new Framework.init(input)
	}
	Framework.prototype = {
		create: function() {
			// remove hash as not required
			const templateName = this.template.replace('#', '')
			// get the template
			const template = document.getElementById(templateName).innerHTML
			// update placeholder with data
			this.template = template.replace(/{{title}}/g, this.data.title)
			return this
		},
		render: function(input) {
			let content = document.querySelector(input)
			content.innerHTML = this.template
			this.addEvents()
			return this
		},
		addEvents: function() {
			const clear = document.querySelector('[data-event="click:clearTitle"]')
			clear.addEventListener('click', this.clearTitle)
			const keyUp = document.querySelector('[data-event="keyup:titleChanged"]')
			keyUp.addEventListener('keyup', this.titleChanged)
		}
	}
	Framework.init = function(input) {
		let self = this
		self.template = input.template || ''
		self.data = input.data || ''
		self.clearTitle = input.clearTitle || ''
		self.titleChanged = input.titleChanged || ''
	}

	Framework.init.prototype = Framework.prototype
	global.Framework = Framework
})(window)
