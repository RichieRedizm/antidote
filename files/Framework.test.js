import Framework from './Framework.js'

const input = {
  template: '#header-template',
  data: {
    title: 'Antidote'
  },
  clearTitle: function(e) {
    document.querySelector('input').value = ''
    document.querySelector('h1').innerHTML = ''
  },
  titleChanged: function(e) {
    document.querySelector('h1').innerHTML = e.target.value
  }
}

test('if given an some input, Framework object is created with defaults', () => {
  expect(
    Framework({
      data: {
        title: 'Antidote'
      }
    })
  ).toMatchSnapshot()
})

test('if given no input, Framework init returns null', () => {
  expect(Framework()).toMatchSnapshot()
})

test('if given correct input, Framework object is created', () => {
  expect(Framework(input)).toMatchSnapshot()
})
