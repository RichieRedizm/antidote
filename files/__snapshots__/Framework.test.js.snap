// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if given an some input, Framework object is created with defaults 1`] = `
Object {
  "clearTitle": "",
  "data": Object {
    "title": "Antidote",
  },
  "template": "",
  "titleChanged": "",
}
`;

exports[`if given correct input, Framework object is created 1`] = `
Object {
  "clearTitle": [Function],
  "data": Object {
    "title": "Antidote",
  },
  "template": "#header-template",
  "titleChanged": [Function],
}
`;

exports[`if given no input, Framework init returns null 1`] = `Object {}`;
